pkg/ddc/alluxio/cache.go

```diff
@@ -17,94 +17,74 @@ package alluxio
 
 import (
 	"fmt"
-	"regexp"
-	"strings"
 	"time"
 
+	"github.com/fluid-cloudnative/fluid/api/v1alpha1"
 	"github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations"
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"github.com/pkg/errors"
+)
+
+var (
+	cachedPercentageFormat = "%.1f%%"
 )
 
 // queryCacheStatus checks the cache status
 func (e *AlluxioEngine) queryCacheStatus() (states cacheStates, err error) {
-	summary, err := e.reportSummary()
+	// get alluxio fsadmin report summary
+	summary, err := e.GetReportSummary()
 	if err != nil {
 		e.Log.Error(err, "Failed to get Alluxio summary when query cache status")
 		return states, err
 	}
-	strs := strings.Split(summary, "\n")
-	for _, str := range strs {
-		str = strings.TrimSpace(str)
-		if strings.HasPrefix(str, SUMMARY_PREFIX_TOTAL_CAPACITY) {
-			totalCacheCapacityAlluxio, _ := utils.FromHumanSize(strings.TrimPrefix(str, SUMMARY_PREFIX_TOTAL_CAPACITY))
-			// Convert Alluxio's binary byte units to Fluid's binary byte units
-			// e.g. 10KB -> 10KiB, 2GB -> 2GiB
-			states.cacheCapacity = utils.BytesSize(float64(totalCacheCapacityAlluxio))
-		}
-		if strings.HasPrefix(str, SUMMARY_PREFIX_USED_CAPACITY) {
-			usedCacheCapacityAlluxio, _ := utils.FromHumanSize(strings.TrimPrefix(str, SUMMARY_PREFIX_USED_CAPACITY))
-			// Convert Alluxio's binary byte units to Fluid's binary byte units
-			// e.g. 10KB -> 10KiB, 2GB -> 2GiB
-			states.cached = utils.BytesSize(float64(usedCacheCapacityAlluxio))
-		}
+
+	if len(summary) == 0 {
+		return states, errors.New("Alluxio summary is empty")
 	}
 
+	// parse alluxio fsadmin report summary
+	states = e.ParseReportSummary(summary)
+
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
 	if err != nil {
-		e.Log.Info("Failed to get dataset when query cache status")
+		e.Log.Error(err, "Failed to get dataset when query cache status")
 		return states, err
 	}
 
-	// `dataset.Status.UfsTotal` probably haven't summed, in which case we won't compute cache percentage
-	if dataset.Status.UfsTotal != "" && dataset.Status.UfsTotal != METADATA_SYNC_NOT_DONE_MSG {
-		usedInBytes, _ := utils.FromHumanSize(states.cached)
-		ufsTotalInBytes, _ := utils.FromHumanSize(dataset.Status.UfsTotal)
-		states.cachedPercentage = fmt.Sprintf("%.1f%%", float64(usedInBytes)/float64(ufsTotalInBytes)*100.0)
-	}
+	e.patchDatasetStatus(dataset, &states)
 
-	states.cacheHitStates = e.getCacheHitStates()
+	states.cacheHitStates = e.GetCacheHitStates()
```

Comment:
Same question.

Reply:
because the current version of `Convey` cannot mock private functions